crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,523 @@
+
+            let futures: Result<Vec<Vec<u8>>, _> = join_all(
+                [
+                    &["--version"][..],
+                    &["config", "--get", "core.fsmonitor"][..],
+                    &["config", "--get", "core.untrackedcache"][..],
+                ]
+                .into_iter()
+                .map(|args| async move {
+                    // get the current setting
+                    let stdout = Stdio::piped();
+
+                    let command = Command::new("git")
+                        .args(args)
+                        .stdout(stdout)
+                        .spawn()
+                        .expect("too many processes"); // this can only fail if we run out of processes on unix
+
+                    command.wait_with_output().await.map(|d| d.stdout)
+                }),
+            )
+            .await
+            .into_iter()
+            .collect(); // transpose
+
+            Self::log_line(&chan, "Collecting metadata".to_string()).await;
+
+            match futures.as_ref().map(|v| v.as_slice()) {
+                Ok([version, fsmonitor, untrackedcache]) => {
+                    let version = String::from_utf8_lossy(version);
+                    let Some(version) = version.trim().strip_prefix("git version ") else {
+                        Self::failed(chan, "Failed to get git version".to_string()).await;
+                        return;
+                    };
+
+                    let (major, rest) = version.split_once('.').expect("semver");
+                    let (minor, _) = rest.split_once('.').expect("semver");
+
+                    let major = major.parse::<u32>().expect("int");
+                    let minor = minor.parse::<u32>().expect("int");
+
+                    if major == 2 && minor < 37 || major == 1 {
+                        Self::failed(
+                            chan,
+                            format!(
+                                "Git version {} is too old, please upgrade to 2.37 or newer",
+                                version
+                            ),
+                        )
+                        .await;
+                        return;
+                    } else {
+                        Self::log_line(
+                            &chan,
+                            format!("Using supported Git version - {}", version).to_string(),
+                        )
+                        .await;
+                    }
+
+                    let fsmonitor = String::from_utf8_lossy(fsmonitor);
+                    let untrackedcache = String::from_utf8_lossy(untrackedcache);
+
+                    if fsmonitor.trim() != "true" || untrackedcache.trim() != "true" {
+                        Self::log_line(&chan, "Git FS Monitor not configured".to_string()).await;
+                        Self::log_line(&chan, "For more information, see https://turbo.dev/repo/docs/reference/command-line-reference/scan#fs-monitor".to_string()).await;
```

Comment:
I don't think we own `turbo.dev` anymore
```suggestion
                        Self::log_line(&chan, "For more information, see https://turbo.build/repo/docs/reference/command-line-reference/scan#fs-monitor".to_string()).await;
```